### DISCUSSION TITLE
Embedding Python module in a Rust application

### DISCUSSION THREAD

**Discussion by Krahos** [indygreg/PyOxidizer#596] [Q&A]
Hello everyone, I have an existing Rust application and I should call a class declared and defined in a Python file. This class has dependencies installed with pip. I successfully tried with pyo3 which works for a file without python dependencies, but they are suggesting PyOxidizer to embed all the dependencies in the executable coming out of cargo build. I'm struggling to understand how to do this, could you provide a simple example about it?

**Comment by AceofSpades5757** [comment]
It looks like this should help shed some light on the limitations of `pyembed`.

<https://pyoxidizer.readthedocs.io/en/latest/pyembed_extension_modules.html>

It seems like only extension modules can be added to the binary at this time. We should be able to add to `sys.path`, but I'm having issues with a `STATUS_ACCESS_VIOLATION` on Windows. Hopefully, you have better luck.
